api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -215,6 +234,11 @@ type VirtualMachinePublishRequestStatus struct {
 	// +optional
 	StartTime metav1.Time `json:"startTime,omitempty"`
 
+	// Attempt represents the information about the attempted requests.
+	//
+	// +optional
+	Attempts VirtualMachinePublishRequestAttempts `json:"attempt,omitempty"`
```

Comment:
- Please update the JSON field name to `attempts`
- This should _probably_ be  type `int64` and tracking the last attempt time should likely occur as part of the conditions. Neither [PodStatus](https://pkg.go.dev/k8s.io/api/core/v1#PodStatus) or [ContainerStatus](https://pkg.go.dev/k8s.io/api/core/v1#ContainerStatus) tracks the last container-pull attempt time as part of the status, but rather as part of the events. Thus I do not know if there is precedent. I might even drop the `Attempts` entirely and simply emit the attempt count as an event to make this similar to pulling a container image.

Reply:
If you do keep this, let's inline it directly into status, ex:

```golang
type VirtualMachinePublishRequestStatus struct {

    // ... ellided ...

    // Attempts represents the number of times the request has been attempted.
    //
    // +optional
    Attempts int64 `json:"attempts,omitempty"`

    // LastAttemptTime represents the time when the latest request was sent.
    //
    // +optional
    LastAttemptTime metav1.Time `json:"lastAttemptTime,omitempty"`
}
```

Still, I wonder if the last attempt time should be a condition?

Reply:
Synced up offline. Will keep `Attempts` and `lastAttemptTime` and incline them directly into status. Resolve this comment. 

`Attempts` are required as we need a field to generate different actIDs when sending publish requests. This field should not be immutable to users.